pymc3/distributions/distribution.py

```diff
@@ -114,10 +114,10 @@ def __init__(self, shape_supp, shape_ind, shape_reps, bcast, dtype, testval=None
             A transform function
         """
 
-        self.shape_supp = shape_supp
-        self.shape_ind = shape_ind
-        self.shape_reps = shape_reps
-        self.shape = shape_supp + shape_ind + shape_reps
+        self.shape_supp = T.cast(T.as_tensor_variable(shape_supp, ndim=1), 'int64')
```

Comment:
`T.cast` causes problems when `T.get_vector_length` is called on these shapes.  Surprising since `T.cast` is an `Elemwise` operation that doesn't affect shape/length.  Maybe we should ask the Theano guys about this/`T.get_vector_length`?